server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -693,7 +694,7 @@ private Set<Index> resolveSystemIndicesToDelete(ClusterState currentState, Set<S
     static DataStream updateDataStream(DataStream dataStream, Metadata.Builder metadata, RestoreSnapshotRequest request) {
         String dataStreamName = dataStream.getName();
         if (request.renamePattern() != null && request.renameReplacement() != null) {
-            dataStreamName = dataStreamName.replaceAll(request.renamePattern(), request.renameReplacement());
+            dataStreamName = dataStreamName.replaceAll(request.renamePattern(), Matcher.quoteReplacement(request.renameReplacement()));
```

Comment:
Is changing it on this level going to break snapshot restore with a replacement pattern?

I wonder if we should instead quote it at:
https://github.com/elastic/elasticsearch/blob/f4e3113ad00541f084b69664c6476c8fcf737386/x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java#L188

I do not think we ever accept follower index in a put follower api as a regexp and that way we are not going to change existing snapshot restore logic.

Reply:
When I was looking at this last week, I thought it was a bug that all name replacements were treated as regex. I assumed that those were supposed to be string literals. But now I see that we support match groups for normal restores, so I changed this to only impact CCR.